src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side is a positive value');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`The longest side of a triangle cannot be greater
+        than the sum of other two`);
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const { a, b, c } = this;
+    const halfPerimeter: number = (a + b + c) / 2;
+    const area: number = Math.sqrt(halfPerimeter * (halfPerimeter - a
```

Comment:
```suggestion
    const halfPerimeter = (a + b + c) / 2;
    const area = Math.sqrt(halfPerimeter * (halfPerimeter - a
```
typescript already knows that it is numbers